### PR TITLE
[Stream] use huggingface standard generation for tnx

### PR DESCRIPTION
**Pull request by lanking520**
## Description ##

Add TNX streaming generation for HuggingFace No-KV interface

### CODE REVIEW

File: engines/python/setup/djl_python/streaming_utils.py [hunk 1]
```diff
@@ -23,10 +23,10 @@ def get_stream_generator(execution_engine: str):
         ## execution_engine passed to this function is not the same engine specified in serving.properties
         ## in djl-serving. For e.g Accelerate and neuronx use Python as the engine serving.properties
         ## The engine here refers to backend model parallel framework.
-        if execution_engine in {"DeepSpeed", "Accelerate"}:
+        if execution_engine in {
```
Comment (frankfliu) [inline engines/python/setup/djl_python/streaming_utils.py:26]: should this be a list ["DeepSpeed", "Accelerate", "transformers-neuronx"] ?

File: engines/python/setup/djl_python/streaming_utils.py [hunk 2]
```diff
@@ -55,6 +55,12 @@ def _hf_model_stream_generator(model, tokenizer, inputs, **kwargs):
                                           dtype=torch.long,
                                           device=input_ids.device)
         stop_generation = False
+        engine = None
+        if "engine" in kwargs.keys():
+            engine = kwargs["engine"]
+
+        if engine and "transformers-neuronx" == engine:
```
Comment (frankfliu) [inline engines/python/setup/djl_python/streaming_utils.py:62]:
```suggestion
        if "transformers-neuronx" == engine:
```

File: engines/python/setup/djl_python/streaming_utils.py [hunk 3]
```diff
@@ -83,10 +89,17 @@ def _hf_model_stream_generator(model, tokenizer, inputs, **kwargs):
...
-                                        attention_mask=attention_mask_curr,
-                                        past_key_values=past_key_values,
-                                        use_cache=True)
+                model_inputs = {
+                    "input_ids": input_ids,
+                    "attention_mask": attention_mask_curr,
+                    "past_key_values": past_key_values,
+                    "use_cache": True
+                }
+                if engine and "transformers-neuronx" == engine:
```
Comment (frankfliu) [inline engines/python/setup/djl_python/streaming_utils.py:98]:
```suggestion
                if "transformers-neuronx" == engine:
```

File: tests/integration/llm/sagemaker-endpoint-tests.py [hunk 4]
```diff
@@ -129,9 +129,7 @@
     }
 }
 
-ENGINE_TO_METRIC_CONFIG_ENGINE = {
```
Comment (frankfliu) [inline tests/integration/llm/sagemaker-endpoint-tests.py:132]: This has be changed, you can rebased master

File: engines/python/setup/djl_python/streaming_utils.py (hunk 1 above)
Comment (lanking520) [reply engines/python/setup/djl_python/streaming_utils.py:26]: A set will be better?

**Pull request merged**